docs/api.md

```diff
@@ -48,6 +48,37 @@ Kubernetes also stores its serialized state (currently in [etcd](https://coreos.
 
 Kubernetes itself is decomposed into multiple components, which interact through its API.
 
+## Adding APIs to Kubernetes
+Every API that is added to Kubernetes carries with it increased cost and complexity for all parts of the Kubernetes ecosystem.  New APIs imply new code to maintain,
+new tests that may flake, new documentation that users are required to understand, increased cognitive load for kubectl users and many other incremental costs.
+
+Of course, the addition of new APIs also enables new functionality that empowers users to simply do things that may have been previously complex, costly or both.
+
+Given this balance between increasing the complexity of the project, vs the reduction of complexity in user actions, we have set out to set up a set of criteria
+by which we as a development community decide when an API should be added to the set of core Kubernetes APIs.
```

Comment:
I would soften this a little, e.g. "we have set out to set up a set of criteria that will help to guide how we as a development community..."